pkg/controllers/management/eks/eks_cluster_handler.go

```diff
@@ -480,6 +485,27 @@ func (e *eksOperatorController) generateAndSetServiceAccount(cluster *mgmtv3.Clu
 	return e.clusterClient.Update(cluster)
 }
 
+func (e *eksOperatorController) canUsePublicAPIEndpoint(cluster *mgmtv3.Cluster) (string, *bool) {
+	var publicAccess bool
+	accessToken, err := e.getAccessToken(cluster)
+	if err != nil {
+		return "", nil
+	}
+
+	netDialer := net.Dialer{
+		Timeout:   30 * time.Second,
+		KeepAlive: 30 * time.Second,
+	}
+
+	serviceToken, err := generateSAToken(cluster.Status.APIEndpoint, cluster.Status.CACert, accessToken, netDialer.DialContext)
+
+	if err == nil {
+		publicAccess = true
```

Comment:
Can we check for a specific error, like a timeout? otherwise maybe we just need to retry, in which case maybe this function should just return the err so the parent function can handle it.

Reply:
If Rancher can't contact the API because it is private-only, would this error always be a timeout?

Reply:
It's a `net.DNSError`, and it is handled more gracefully.